libs/designsystem/src/lib/components/dropdown/dropdown.component.ts

```diff
@@ -1,30 +1,40 @@
 import {
+  AfterContentChecked,
+  AfterViewInit,
+  ChangeDetectorRef,
   Component,
-  HostBinding,
-  Input,
-  TemplateRef,
   ContentChild,
-  HostListener,
-  ElementRef,
   ContentChildren,
-  ViewChildren,
-  QueryList,
-  ViewChild,
-  AfterContentChecked,
-  Renderer2,
-  Output,
+  ElementRef,
   EventEmitter,
-  OnDestroy,
   forwardRef,
-  AfterViewInit,
-  ChangeDetectorRef,
+  HostBinding,
+  HostListener,
+  Input,
+  OnDestroy,
+  Output,
+  QueryList,
+  Renderer2,
+  TemplateRef,
+  ViewChild,
+  ViewChildren,
 } from '@angular/core';
 import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
 
-import { ListItemTemplateDirective } from '../list/list.directive';
-import { ItemComponent } from '../item/item.component';
 import { CardComponent } from '../card/card.component';
+import { ItemComponent } from '../item/item.component';
+import { ListItemTemplateDirective } from '../list/list.directive';
 
+export enum openState {
+  closed,
+  opening,
+  open,
+}
```

Comment:
Could you comment on why this is better than the string union type? 🤔

Reply:
Sure. There are a number of things :)
1. I think it is much cleaner and easier to reason about to have explicit types like this in stead of just strings
2. Reusability, the string-union types need to be declared every time whereas the enums can live in a central place an be consistent across the solution (ie. attentionLevel and sizes are excellent candidates for that)
3. Better tooling support - type a `.` and get all options
4. String literals are objectively bad :)
5. Union types is IMO an anti-pattern (though some times necessary), as it inherently undermines the main reason you would adopt typescript over vanilla JS.
6. The string-union type IS essentially an emulation of an enum rather than a union-type - IMO better to use the proper language construct